### DISCUSSION TITLE
Is there a RISC-V port of pyOCD?

### DISCUSSION THREAD

**Discussion by RobertHenry6bev** [pyocd/pyOCD#1224] [Q&A] [answered]
Is anybody working on or aware of a riscv port of pyOCD?

**Comment by kammce** [comment]
+1 question/request

**Comment by cederom** [comment]
Would be nice to have but I guess we need to make that happen :-)

**Comment by flit** [comment, marked as answer]
There is not. At the moment pyocd only supports Arm M-profile cores and CoreSight based devices. (A-profile and maybe R-profile cores will come at some point, although nobody's really asking for it.)

You're welcome to submit changes to support RISC-V! Now that pyocd is an independent project (as of Sep 2020) and not owned by Arm, we can support other architectures. I won't and can't implement it myself, because I'm Arm employee and that would violate my employment contract (competing interests). However, I will be happy to work on structural changes and provide guidance and recommendations for others. (And to be clear, I'd like to see it happen.)

Be aware that the current pyocd design needs to be changed to support multiple architectures. In the background, I've actually been planning how a multi-architecture pyocd design would look, and the changes will be made (most of these changes are also needed for A-profile), but it's still quite a ways off.

Another thing to keep in mind is that any additions need to be well supported with a maintainer (can't be me, for above reasons) and follow the pyocd design principles of being (or attempting to be 😉) easy to use while supporting a wide range of devices, and taking advantage of the RISC-V ecosystem.

**Reply by cederom** [reply]
I have created LibSWD [1] some time ago that provided platform and tool independent SWD transport and ARM Cortex-M DAP access. I am using pyOCD everyday on FreeBSD OS and I am also interested in RISC-V architecture (have MAX78000 and ESP32-C3 boards already). If there is a company out there open for contract I would be more than happy to do the job :-)

[1] https://github.com/CeDeROM/LibSWD
